src/pip/_internal/resolution/resolvelib/resolver.py

```diff
@@ -122,7 +124,10 @@ def resolve(self, root_reqs, check_supported_wheels):
             )
 
         except ResolutionImpossible as e:
-            error = self.factory.get_installation_error(e, constraints)
+            error = self.factory.get_installation_error(
+                cast("ResolutionImpossible[Requirement, Candidate]", e),
+                constraints,
+            )
```

Comment:
Is there a better way to handle the type of `e`? `ResolutionImpossible` is a generic exception, but I’m not sure how to specify the type variables here.

Reply:
This is fine.